### PR TITLE
http/endpoint: extract common pipe code

### PR DESCRIPTION
**Pull request by alxndrsn**
defaultResultWriter and odataJsonWriter use identical code for piping results to response.  This commit extracts the shared code into its own function.

Follow-up to #1110

### CODE REVIEW

File: lib/http/endpoint.js [hunk 1]
```diff
@@ -192,20 +192,28 @@ const endpointBase = ({ preprocessor = noop, before = noop, resultWriter, errorW
...
   if (result instanceof PartialPipe) {
     result.streams.at(-1).on('error', streamErrorHandler(response));
     result.with(response).pipeline((err) => next?.(err));
+    return true;
   } else if (result.pipe != null) {
     result.on('error', streamErrorHandler(response));
     pipeline(result, response, (err) => err && next?.(err));
+    return true;
+  }
+};
```
Comment (matthew-white) [inline lib/http/endpoint.js:209]: I know `undefined` is falsy, but I think it'd be nice if the function returned `false` if it doesn't return `true`.

File: lib/http/endpoint.js [hunk 2]
```diff
@@ -192,20 +192,28 @@ const endpointBase = ({ preprocessor = noop, before = noop, resultWriter, errorW
 // ENDPOINT FORMAT SPECIALIZATIONS
 
 ////////////////////////////////////////
```
Comment (matthew-white) [inline lib/http/endpoint.js:194]: I think this line comment goes together with the line comment on the next line, so the two should be moved together. I think the two lines should come before `defaultResultWriter()`/`defaultErrorWriter()`/`defaultEndpoint()`, similar to how there are line comments above `odataJsonWriter()`/`odataXmlErrorWriter()`/`odataJsonEndpoint()`/etc.

**Review by matthew-white — APPROVED** [review]
I know it's been a while since this PR was pushed, but I just took a look, and the changes make sense to me. 💯

File: lib/http/endpoint.js (hunk 2 above)
Comment (alxndrsn) [reply lib/http/endpoint.js]: Could we just remove the comment completely?

File: lib/http/endpoint.js (hunk 1 above)
Comment (alxndrsn) [reply lib/http/endpoint.js]: Added.

File: lib/http/endpoint.js (hunk 2 above)
Comment (matthew-white) [reply lib/http/endpoint.js]: Sounds fine to me, I'm not attached to it. :+1:

**Pull request merged**